server/aws-lsp-codewhisperer/src/language-server/inline-completion/codeWhispererServer.ts

```diff
@@ -592,6 +592,14 @@ export const CodewhispererServerFactory =
             logging.log('Recommendation failure: ' + error)
             emitServiceInvocationFailure(telemetry, session, error)
 
+            if (error instanceof AmazonQError) {
+                throw error
+            }
+
+            if (hasConnectionExpired(error)) {
+                throw new AmazonQServiceConnectionExpiredError(getErrorMessage(error))
```

Comment:
![image](https://github.com/user-attachments/assets/d95cedd8-fa7e-4596-9022-079179959337)
Some of the missing code involves wrapping the exception and returning the standardized `ResponseError` (which is part of the standard LSP Protocol). Is ResponseError being returned from somewhere higher in the call chain, or does this code require some sort of wrapping?

Reply:
thanks! I added the translate logic back in latest commit, not sure why it was originally nested but I think this should have the same effect?